### PR TITLE
Install CMake in /usr/local/bin in prowbazel images.

### PR DESCRIPTION
**Pull request by PiotrSikora**
While there, fix a bunch of non-working mkdir calls.

Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

File: docker/prowbazel/Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ RUN /tmp/istio_tmp/scripts/linux-install-software \
 VOLUME /var/lib/docker
 EXPOSE 2375
 
-ENV PATH /usr/local/go/bin:/usr/local/cmake/bin:/usr/lib/google-cloud-sdk/bin:/opt/go/bin:${PATH}
+ENV PATH /usr/local/go/bin:/usr/lib/google-cloud-sdk/bin:/opt/go/bin:${PATH}
 
 # Create bootstrap user
 ENV HOME /home/bootstrap
```
Comment (fejta) [inline docker/prowbazel/Dockerfile:27]: I'd recommend combining these into a single ENV layer

File: scripts/tools/linux-install-bazel [hunk 2]
```diff
@@ -19,7 +19,7 @@ function update_bazel() {
   local bazel_current="$(bazel version | grep -oh 'Bazelisk version: v[0-9]*\.[0-9]*')" \
     || bazel_current='none'
 
-  if [[ -d "${BAZELISK_DIRECTORY}" ]]; then
+  if [[ ! -d "${BAZELISK_DIRECTORY}" ]]; then
     mkdir -p "${BAZELISK_DIRECTORY}"
```
Comment (fejta) [inline scripts/tools/linux-install-bazel]: why not just run `mkdir -p`? It has the exact same behavior.

**Review by fejta — APPROVED** [review]
/hold

File: docker/prowbazel/Dockerfile (hunk 1 above)
Comment (PiotrSikora) [reply docker/prowbazel/Dockerfile:27]: Those are separate logical steps (and unrelated change, so I'd rather not do it in this PR).

File: scripts/tools/linux-install-bazel (hunk 2 above)
Comment (PiotrSikora) [reply scripts/tools/linux-install-bazel]: Good catch, thanks! Fixed.

File: docker/prowbazel/Dockerfile (hunk 1 above)
Comment (fejta) [reply docker/prowbazel/Dockerfile:27]: Hmm, well I was all excited to link to the best practices, but it appears that this has changed in the past few years:

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

```
In older versions of Docker, it was important that you minimized the number of layers in your images to ensure they were performant. The following features were added to reduce this limitation:

Only the instructions RUN, COPY, ADD create layers. 
```

**Review by fejta — APPROVED** [review]
/hold

**Comment by fejta** [comment]
/hold cancel
I mean

**Comment by PiotrSikora** [comment]
@howardjohn @sdake @geeknoid @duderino @linsun @ericvn this is blocked on `CODEOWNERS` for `/docker/*`, could one of you approve?

**Review by howardjohn — APPROVED** [review]

**Comment by PiotrSikora** [comment]
@fejta is this image going to be built and pushed automatically or is it a manual process? I don't see any postsubmits running.

**Comment by howardjohn** [comment]
it is manual, but we should change that

**Comment by fejta** [comment]
We have recipes for doing this, which we can start using now that there is a trusted cluster.

**Pull request merged**